tfhe/src/shortint/engine/mod.rs

```diff
@@ -163,12 +165,18 @@ impl ShortintEngine {
         // Create the accumulator
         let mut accumulator_u64 = vec![0_u64; server_key.bootstrapping_key.polynomial_size().0];
 
+        // Tracking the max value of the function to define the degree later
+        let mut max_value = 0;
+
         // This accumulator extracts the carry bits
         for i in 0..modulus_sup {
             let index = i * box_size;
             accumulator_u64[index..index + box_size]
                 .iter_mut()
                 .for_each(|a| *a = f(i as u64) * delta);
+            if f(i as u64) > max_value {
+                max_value = f(i as u64);
+            }
```

Comment:
this should work ok:

```rust
max_value = max_value.max(f(i as u64))
```

Reply:
I read that the max for Iterator was slower due to the Option it returns. Do you have an opinion on that?

Reply:
here what I propose is not on an iterator, max_value is e.g. a u64, so max_value.max(other) is just comparing two values and updating max value, basically your piece of code but f is potentially only evaluated once instead of twice (though normally the compiler can see there is no side effect and should call f only once)